analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AssignmentExpressionSyntaxExtensionsTests.cs

```diff
@@ -27,38 +27,492 @@ namespace SonarAnalyzer.UnitTest.Extensions
+                    new
+                    {
+                        Left = WithIdentifier("y"),
+                        Right = WithToken("2"),
+                    },
+                });
+
+        [TestMethod]
+        public void MapAssignmentArguments_NestedDeconstructionAssignment() =>
+            AssertMapAssignmentArguments("var (a, (b, c), d) =  (1, (2, 3), 4);", new[]
+                {
+                    new
+                    {
+                        Left = WithIdentifier("a"),
+                        Right = WithToken("1"),
+                    },
+                    new
+                    {
+                        Left = WithIdentifier("b"),
+                        Right = WithToken("2"),
+                    },
+                    new
+                    {
+                        Left = WithIdentifier("c"),
+                        Right = WithToken("3"),
+                    },
+                    new
+                    {
+                        Left = WithIdentifier("d"),
+                        Right = WithToken("4"),
+                    },
+                });
+
+        [TestMethod]
+        public void MapAssignmentArguments_MisalignedDeconstructionAssignment() =>
+            AssertMapAssignmentArguments("var (a, (b, c, d)) =  (1, (2, 3), 4);", new[]
+                {
+                    new
+                    {
+                        Left = new
+                        {
+                            Designation = new
+                            {
+                                Variables = new object[]
+                                {
+                                    WithIdentifier("a"),
+                                    new
+                                    {
+                                        Variables = new[]
+                                        {
+                                            WithIdentifier("b"),
+                                            WithIdentifier("c"),
+                                            WithIdentifier("d"),
+                                        }
+                                    },
+                                },
+                            },
+                        },
+                        Right = new
+                        {
+                            Arguments = new object[]
+                            {
+                                new { Expression = WithToken("1") },
+                                new
+                                {
+                                    Expression = new
+                                    {
+                                        Arguments = new object[]
+                                        {
+                                            new { Expression = WithToken("2") },
+                                            new { Expression = WithToken("3") },
+                                        },
+                                    },
```

Comment:
This part repeats 7x. Let's do `WithArgumentTokens("2", "3")` (or something like that) using `params`